src/connectdlg.cpp

```diff
@@ -524,9 +529,20 @@ void CConnectDlg::OnServerAddrEditTextChanged ( const QString& )
 
 void CConnectDlg::OnCustomCentralServerAddrChanged()
 {
-    // only update list if the custom server list is selected
+    UpdateDirectoryServerComboBox();
+    // only update list if a custom server list is selected
```

Comment:
Can you
1. Add a "TODO" for only updating the list of the actual directory server selected changed (i.e. remember where we were and see if we can find it in the new list).
2. Drop the other comments here.

I'm okay with this approach as a first cut, but I'd like to see an issue raised - and linked from the TODO - covering point 1.

Reply:
Yes I agree with you about adding a TODO but I think a discussion is needed around the broad topic of user interface for finding and connecting to servers, including:
- directory management
- server connection history
- server connection favourites/bookmarks
- ad-hoc server connection
- parental controls

Maybe there is already an appropriate discussion open ?  #1774 seems relevant.

I would like to propose a mock-up to address that part of the UI in a way that accommodates all types of users.